cachi2/core/package_managers/rpm/main.py

```diff
@@ -241,6 +251,8 @@ def _generate_sbom_components(
     """Fill the component list with the package records."""
     components: list[Component] = []
     for file_path, file_metadata in files_metadata.items():
+        if not str(file_path).endswith(".rpm"):
```

Comment:
`file_path` should be a `Path`, you could get rid of a conversion here by checking its suffix directly:
```suggestion
if file_path.suffix != ".rpm":
```

This also is starting to get a bit complicated. I suggest shifting this to a helper called `is_rpm_metadata()` or something similar, otherwise it could be a bit unclear why non-rpm files are expected.